### PR TITLE
Some cache_bench enhancements

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: ... used in validating some HyperClockCache development in progress.

* Revamp the "populate cache" step to avoid redundant insertions (very rare in practice) and more consistently approach the desired resident_ratio while maintaining appropriate skew (still not perfect).
* Track and print hit ratio on lookups, to ensure a fair comparison is happening between implementations etc.
* Add an option to disable tracking and printing histograms (lots of output)
* Add an option to specify a random seed (for more reproducibility)
* Remove confusing/redundant "-skewed" option

Uses BitwiseAnd from #11660 (tested there)

Test Plan: manual

### CODE REVIEW

**Review by jowlyzhang — APPROVED** [review]
LGTM, thanks for the PR!

File: cache/cache_bench_tool.cc [hunk 1]
```diff
@@ -960,11 +1014,14 @@ int cache_bench_tool(int argc, char** argv) {
     exit(1);
   }
 
+  if (FLAGS_seed == 0) {
+    FLAGS_seed = static_cast<uint32_t>(port::GetProcessID());
```
Comment (jowlyzhang) [inline cache/cache_bench_tool.cc:1018]: Process id is quite random too, I'm wondering why this helps with reproducibility.

File: cache/cache_bench_tool.cc [hunk 2]
```diff
@@ -333,13 +332,50 @@ class CacheBench {
...
+      }
+      keys_since_last_not_found = 0;
+
+      Status s =
+          cache_->Insert(key, createValue(rnd), &helper1, FLAGS_value_bytes);
       assert(s.ok());
+
+      handle = cache_->Lookup(key);
+      if (!handle) {
+        fprintf(stderr, "Failed to lookup key just inserted.\n");
```
Comment (akankshamahajan15) [inline cache/cache_bench_tool.cc:362]: Should we print the key as well in the error message?

File: cache/cache_bench_tool.cc [hunk 3]
```diff
@@ -333,13 +332,50 @@ class CacheBench {
...
+
+      size_t occ = cache_->GetOccupancyCount();
+      if (occ > max_occ) {
+        max_occ = occ;
+        inserts_since_max_occ_increase = 0;
+      } else {
+        ++inserts_since_max_occ_increase;
+      }
     }
+    printf("Population complete (%zu entries, %g average charge)\n", max_occ,
```
Comment (akankshamahajan15) [inline cache/cache_bench_tool.cc:377]: Just for my understanding. What's charge?

**Review by akankshamahajan15 — APPROVED** [review]
Added few comments for my understanding. LGTM

File: cache/cache_bench_tool.cc (hunk 1 above)
Comment (pdillinger) [reply cache/cache_bench_tool.cc:1018]: `DEFINE_uint32(seed, 0, "Hashing/random seed to use. 0 = choose at random");`

So by default, it's random(-ish), but we print the seed needed to reproduce that run if needed.

File: cache/cache_bench_tool.cc (hunk 3 above)
Comment (pdillinger) [reply cache/cache_bench_tool.cc:377]: As a generic cache, "charge" is a weight associated with an entry such that the cache keeps the total weight below the configured capacity. As a block cache, the weight is the estimated memory usage of the entry in bytes, to keep total block cache memory below the configured limit.

File: cache/cache_bench_tool.cc (hunk 2 above)
Comment (pdillinger) [reply cache/cache_bench_tool.cc:362]: The keys are essentially random, so are meaningless unless you're running it in a debugger. And if you're debugging in a debugger, there's a lot more you'll want to look at anyway, so I don't think it's helpful.

**Pull request closed without merging**